hilti/runtime/include/fiber.h

```diff
@@ -44,7 +105,7 @@ namespace detail {
  */
 class Fiber {
 public:
-    Fiber();
+    Fiber(bool main_fiber = false);
```

Comment:
Is there a hard reason why the main fiber couldn't also use the shared stack? Asking since this leads to some complications in the implementations and I am unsure whether we need it.

Reply:
Yeah, because it already has its own stack that the OS provided. (This looks different now, anyways)